rs/orchestrator/src/catch_up_package_provider.rs

```diff
@@ -209,14 +217,79 @@ impl CatchUpPackageProvider {
     // caller.
     async fn fetch_catch_up_package(
         &self,
-        url: Url,
+        node_id: &NodeId,
+        url: String,
         param: Option<CatchUpPackageParam>,
     ) -> Option<pb::CatchUpPackage> {
-        Agent::new_with_client(self.client.clone(), url, Sender::Anonymous)
-            .query_cup_endpoint(param)
+        let body = Bytes::from(
+            param
+                .and_then(|param| serde_cbor::to_vec(&param).ok())
+                .unwrap_or_default(),
+        );
+
+        let client_config = self
+            .crypto_tls_config
+            .client_config(*node_id, self.registry.get_latest_version())
```

Comment:
Should we use the registry of the CUP here and not the lastest version?

Reply:
We may not have a local CUP at this point, so using the latest version seems correct